python/kserve/pyproject.toml

```diff
@@ -35,7 +35,7 @@ exclude = ["kserve/storage/test"]
 [tool.poetry.dependencies]
 python = ">=3.8,<3.12"
 uvicorn = { version = "^0.19.0", extras = ["standard"] }
-fastapi = "^0.95.0"
+fastapi = "^0.109.0"
 cloudevents = "^1.6.2"
 six = "^1.16.0"
```

Comment:
I think it is better to explicitly add pydantic dependency here.

Reply:
One more thing, I recently found out that fastapi 0.109.0 breaks ray serve. https://github.com/ray-project/ray/pull/42740. So let's keep it as it is for now.

Reply:
Well to upgrade pydantic to v2 we need to upgrade to at least 0.100.0.

https://github.com/tiangolo/fastapi/releases/tag/0.100.0 -- it's probably best to increment it maximally so I'll set it to 0.108.0 as was done in the linked PR.

Reply:
@sivanantha321 I updated the lockfiles with the serving runtimes, added the pydantic 2 syntax to the new models, and added an explicit pydantic dependency.